reverse_proxy_handler_test.cc

```diff
@@ -0,0 +1,100 @@
+#include <string>
+#include <sstream>
+
+#include "gtest/gtest.h"
+#include "reverse_proxy_handler.h"
+#include "request_handler.h"
+#include "response.h"
+#include "request.h"
+#include "config_parser.h"
+
+
+//Test fixture for creating configs and init RequestProxyHandler
+class ReverseProxyHandlerTest : public ::testing::Test {
+protected:
+	bool parseString(const std::string config_string) {
+		std::stringstream config_stream(config_string);
+		return parser.Parse(&config_stream, &out_config);
+	}
+
+	RequestHandler::Status initProxy(std::string prefix){
+		return proxy_handler.Init(prefix,out_config);
+	}
+
+	std::string getPrefix(){
+		return proxy_handler.prefix_;
+	}
+
+	std::string getProtocol(){
+		return proxy_handler.protocol_;
+	}
+
+	std::string getHost(){
+		return proxy_handler.host_;
+	}
+
+	std::string getPath(){
+		return proxy_handler.path_;
+	}
+
+	std::string getUrl(){
+		return proxy_handler.url_;
+	}
+
+	Request TransformIncomingRequest(const Request& request) const{
+		return proxy_handler.TransformIncomingRequest(request);
+	}
```

Comment:
i think he made this class a friend of the proxy handler class? I didn't know you could do that though, and yeah idk if it works either. never tried that